pkg/persistence/object_store_test.go

```diff
@@ -541,6 +542,134 @@ func TestGetDownloadURL(t *testing.T) {
 	}
 }
 
+type objectStoreGetter map[string]velero.ObjectStore
+
+func (osg objectStoreGetter) GetObjectStore(provider string) (velero.ObjectStore, error) {
+	res, ok := osg[provider]
+	if !ok {
+		return nil, errors.New("object store not found")
+	}
+
+	return res, nil
+}
+
+// TestNewObjectBackupStore runs the NewObjectBackupStore constructor and ensures
+// that an ObjectBackupStore is constructed correctly or an appropriate error is
+// returned.
+func TestNewObjectBackupStore(t *testing.T) {
+	tests := []struct {
+		name              string
+		location          *velerov1api.BackupStorageLocation
+		objectStoreGetter objectStoreGetter
+		wantBucket        string
+		wantPrefix        string
+		wantErr           string
+	}{
+		{
+			name:     "location with no ObjectStorage field results in an error",
+			location: &velerov1api.BackupStorageLocation{},
```

Comment:
Good question - I wasn't sure which package I'd want them to go in, and since I was only working on this one test func, I sort of deferred it.

I was also mulling whether we should switch to type-specific functional options instead, as it seems to be the more preferred approach in the Go community. Would be a little bit of a pain to do at this point, but I could get it done.

Reply:
I think it may be useful to have type-specific functional args. Working on `metav1.Object` is useful in some of the tests, like when you have a PVC or PV, but 2 of my PRs now have had to go type-specific anyway because they need types that aren't exposed by the API server, and thus aren't standalone.

It would also remove all those type-check-and-panic lines, so I'm not sure it would even be more lines of code.

Reply:
Call me crazy but part of me thinks we should just throw all of the builder/options code away and go back to struct literals. It's more verbose but maybe just simpler.

Reply:
EDIT: I do think my previous comment is crazy. But, I would like to figure out a consistent style here that works for all scenarios, as we have a couple of different things going on right now.